### PR TITLE
Use scu to avoid unnecessary rendering

### PR DESCRIPTION
**Pull request by lvybriage**
also, use this._isMounted to check the component status.
For issue #1024

### CODE REVIEW

File: src/webui/src/components/trial-detail/Intermeidate.tsx [hunk 1]
```diff
@@ -204,28 +213,73 @@ class Intermediate extends React.Component<IntermediateProps, IntermediateState>
...
-        if (isFilter === true) {
-            const pointVal = this.pointInput !== null ? this.pointInput.value : '';
-            const minVal = this.minValInput !== null ? this.minValInput.value : '';
-            if (pointVal === '' && minVal === '') {
-                this.drawIntermediate(nextProps.source);
+    componentWillReceiveProps(nextProps: IntermediateProps, nextState: IntermediateState) {
+        const { isFilter, filterSource } = nextState;
+        const { whichGraph, source } = nextProps;
+        
+        if (whichGraph === '4') {
```
Comment (yds05) [inline src/webui/src/components/trial-detail/Intermeidate.tsx]: what does '4' mean in this context?

Comment (lvybriage) [reply src/webui/src/components/trial-detail/Intermeidate.tsx]: '4' means that user is looking at intermediate result graph. I'll just render intermediate graph rather then all graphs.

File: src/webui/src/components/TrialsDetail.tsx [hunk 2]
```diff
@@ -241,13 +255,19 @@ class TrialsDetail extends React.Component<{}, TrialDetailState> {
...
                 break;
             case '50':
-                this.setState(() => ({ entriesTable: 50 }));
+                if (this._isMounted) {
+                    this.setState(() => ({ entriesTable: 50 }));
+                }
                 break;
             case '100':
-                this.setState(() => ({ entriesTable: 100 }));
+                if (this._isMounted) {
```
Comment (chicm-ms) [inline src/webui/src/components/TrialsDetail.tsx]: It seems unnecessary to use 3 cases for 20, 50 and 100, these 3 cases can be combined into this.setState(() => ({ entriesTable: value }));

Comment (lvybriage) [reply src/webui/src/components/TrialsDetail.tsx]: Ok, I'll change. Thanks!

**Pull request merged**